Cython/Compiler/Nodes.py

```diff
@@ -3906,31 +3936,24 @@ def generate_keyword_unpacking_code(self, min_positional_args, max_positional_ar
         code.putln('kw_args = PyDict_Size(%s);' % Naming.kwds_cname)
         if self.num_required_args or max_positional_args > 0:
             last_required_arg = -1
-            last_required_posonly_arg = -1
             for i, arg in enumerate(all_args):
                 if not arg.default:
                     last_required_arg = i
-                if arg.pos_only and not arg.default:
-                    last_required_posonly_arg = i
             if last_required_arg < max_positional_args:
                 last_required_arg = max_positional_args-1
-            if max_positional_args > 0:
+            if max_positional_args > num_posonly_args:
                 code.putln('switch (pos_args) {')
             for i, arg in enumerate(all_args[:last_required_arg+1]):
```

Comment:
Maybe like this?
`for i, arg in enumerate(all_args[num_posonly_args:last_required_arg+1], num_posonly_args):`

Reply:
Done